### PR TITLE
[WIP] - path resolution + typing + linting + Readme

### PR DESCRIPTION
**Pull request by nazariyv**
A lot of changes here.

I am opening this up as WIP so that I can discuss with you about a bunch of issues I am having.

First thing to sort out is the modification of package.json in example when I run `yarn start`. For some reason it gets modified and a bunch of stuff / config gets deleted from it. It must be this react-rewired tinkering with it. What is your thinking behind using it?

### CODE REVIEW

**Comment by kremalicious** [comment]
please do not change our linting rules just because you feel like it. Why do you change our `prettierrc`, which of your problems is that solving? This then leads to 45 changed files in one PR which is almost impossible to review. And again, please do not change our use of `npm`.

If you have specific issues do not just change the codebase in a PR, but open up an issue. It seems to me you should work in your own fork, and not try to bring your changes back into the `main` codebase

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-### Changelog
+# Changelog
```
Comment (kremalicious) [inline CHANGELOG.md:1]: this file is autogenerated and should not be touched at all

**Comment by kremalicious** [comment]
and by renaming the `travis.yml` file you also broke all Travis builds for this PR

File: CHANGELOG.md (hunk 1 above)
Comment (nazariyv) [reply CHANGELOG.md:1]: ok. will revert, but this breaks the md linting rules. not a big problem, of course

**Comment by nazariyv** [comment]
ok, will change back. but the convention is also to have the extension: .yaml

**Comment by nazariyv** [comment]
> please do not change our linting rules just because you feel like it. Why do you change our `prettierrc`, which of your problems is that solving? This then leads to 45 changed files in one PR which is almost impossible to review. And again, please do not change our use of `npm`.
> 
> If you have specific issues do not just change the codebase in a PR, but open up an issue. It seems to me you should work in your own fork, and not try to bring your changes back into the `main` codebase. Opening a PR on here means you are intending to get your changes solving some issue into `main`. As it is right now this PR does not look to solve anything except for changing everything to adhere to your personal coding preferences

Will be making minimal changes to just get my app out. Sorry for the disturbance. Will close the PR.

**Comment by nazariyv** [comment]
none of this is my preference

semicolons: https://medium.com/@thakkery/a-practical-style-guide-to-semi-colons-in-es6-c4f63d03a904#:~:text=Even%20Google's%20summarized%20ES6%20style,There%20is%20a%20good%20reason.&text=Include%20semi%2Dcolons%20at%20the,does%20NOT%20contain%20curly%20braces.

.yaml: https://stackoverflow.com/questions/21059124/is-it-yaml-or-yml?rq=1#:~:text=yml%22%20is%20%22the%20file%20extension,and%20is%20much%20more%20common.

README changes follow the .md linting rules

yarn >> npm

All the changes I have made revealed deep issues with types in this project, the main ones mentioned in Discord. But seeing my work isn't valued, I will make minimal changes in my fork to just get the app out

**Comment by kremalicious** [comment]
you have to acknowledge that there is more to all these things than your personal development preferences or one Stack Overflow post. You are contributing to an open source library as an external contributor, not as a project creator or maintainer. We really appreciate those contributions but it's not a contribution if you repeatedly ignore the rules already in place in this project without telling us what your contribution solves. In any other open-source project your PR would have been closed right away, because changing the whole codebase without any clear intention

`yaml` & `yml` are both valid file endings for YAML files, just like it does not matter if you use `jpg` or `jpeg` for a JPEG file. But Travis does not support this and uses `yml`. Which I told you. But both file endings are valid yaml files and I am not aware of any IDE which doesn't know that `yml` is a YAML file.

Linting rules are usually decided within an organization, and that is what we have in place with [eslint-config-oceanprotocol](https://github.com/oceanprotocol/eslint-config-oceanprotocol). Using no semicolons is the standard in the base recommended ESLint config we are extending our rules from. If you think you can just change those rules in one project only, then yes, it's only your personal preference, ignoring the rules set in place by project, team and organisation. Linting rules are there to keep consistency between OUR projects.

MD: what linting rules? We do not have any markdown linting setup in here so again you are using your own personal preferences not rooted in the project you are contributing to. You are welcome to setup `markdownlint` linting in this repo and present your chosen rules for it though. You would still have to find a way to either:

- ignore the CHANGELOG.md
- add a changelog template for `auto-changelog` following those linting rules

Otherwise, as I said, your changes in CHANGELOG.md will be overwritten again on next release, because this is when the changeling is automatically generated.

Yarn: that again is your personal preference. We use `npm` over here and I told you that already. But you still insist that "yarn >> npm" and think your own opinion about that is enough to change it for every participant in this project. That's why I asked you to rather open up issues before putting in work so we can discuss.

> All the changes I have made revealed deep issues with types in this project
Nope, can't confirm. I can only assume your `dist/` folder seems to have an old build before all the latest changes made in last days. So again, start fresh, make sure to run a new `npm run build` before starting the example.

If you still have problems, again, please describe them individually in a new issues for us to look at and discuss. If you say you had typings problems then why do you think the solution involves changing the package manager, and changing linting rules?

**Comment by nazariyv** [comment]
I agree with some of your points, but not all. I do not intend to spend any more time on this discussion

The way this library is written shows a lack of understanding of how Typescript works and at places how Javascript works

**Comment by kremalicious** [comment]
Please describe your problem in an issue. Baseless overconfidently incorrect accusations won't be answered.

**Pull request closed without merging**